dbt/adapters/spark/connections.py

```diff
@@ -154,7 +156,38 @@ def _connection_keys(self) -> Tuple[str, ...]:
         return "host", "port", "cluster", "endpoint", "schema", "organization"
 
 
-class PyhiveConnectionWrapper(object):
+class ConnectionWrapper(ABC):
```

Comment:
I think I like this idea, but I have some questions. How does this class differ from `BaseConnectionManager` in `dbt-core`:
https://github.com/dbt-labs/dbt-core/blob/a1b067c683212bccd9bc10f6e52bcdd168ccf7ec/core/dbt/adapters/base/connections.py#L59
I know we already had `PyhiveConnectionWrapper`; I'm not very familiar with that class, but it also seems specific to `dbt-spark`. This interface looks very generic, like a set of minimal methods that any adapter would need to connect to something. Put another way, if we do need this class in addition to `BaseConnectionManager`, perhaps we should put `ConnectionWrapper` in `dbt-core` and inherit from it here? Is there any reason to not do that?

Reply:
I think the ConnectionWrapper is pretty spark specific for the time being, @Fokko in case we want to add a generic connection wrapper in core in the future can we rename this class to: `SparkConnectionWrapper`?
```suggestion
class SparkConnectionWrapper(ABC):
```

Reply:
Thanks for the review! I agree with Colin, I think it is very Spark-specific, and I don't it doesn't add much value to move this to the core package. I've renamed the class 👍🏻